src/pages/ProcessDetail.tsx

```diff
@@ -78,8 +78,10 @@ function ProcessDetail({ match, query, setQuery }: IProps) {
         if (stepUserInput && localStepUserInput) return;
         const localTabs = localStepUserInput ? ["user_input", "process"] : ["process"];
         const localSelectedTab = localStepUserInput ? "user_input" : "process";
-
-        setProductName(productNameById(processInstance.product, products));
+        const getProductId = (product: string | { product_id: string }) =>
+            typeof product === 'string' ? product : product.product_id;
```

Comment:
It is better to check if the product is a object, since product can also be null, which gave me `annot read properties of null`. It's fine with auto updates, but without auto updates it will stay a blank page.

```suggestion
            typeof product === 'object' ? product.product_id : product;
```